### PR TITLE
[frontend/backend] fix and improve dashboard statistics (#1697)

### LINKED ISSUE #1697: Improve performances on Dashboard statistics

### PR DESCRIPTION
**Pull request by guillaumejparis**
Close #1697

### CODE REVIEW

File: openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java [hunk 1]
```diff
@@ -103,17 +127,17 @@ List<RawInjectExpectation> allGrantedInjectExpectationsFromDate(@Param("from") I
           +
           "FROM exercises " +
           "INNER JOIN injects ON exercises.exercise_id = injects.inject_exercise " +
-          "JOIN injects_expectations ie ON injects.inject_id = ie.inject_id " +
+          "LEFT JOIN injects_expectations ie ON injects.inject_id = ie.inject_id " +
```
Comment (guillaumejparis) [inline openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java:130]: I needed a LEFT JOIN oterhwis in some case I do not have any data in the mitre aray in dashboard

**Comment by RomuDeuxfois** [comment]
MITRE ATT&CK Coverage still empty on my side.
I have one simulation and one atomic testing
image

**Comment by RomuDeuxfois** [comment]
I am not sure, but this card seems to not working
image

File: openbas-front/src/admin/components/Dashboard.tsx [hunk 2]
```diff
@@ -74,38 +69,6 @@ const Dashboard = () => {
...
-  ];
-  const countByCategory = R.countBy((exercise: ExerciseSimple) => exercise?.exercise_category || t('Unknown'), exercisesFromStore);
-  const categoriesLabels: string[] = R.keys(countByCategory).length === 0 ? categoriesLabelsFakeData : R.keys(countByCategory);
-  const categoriesData: number[] = R.values(countByCategory).length === 0 ? categoriesDataFakeData : R.values(countByCategory);
-  const sortByY = R.sortWith([R.descend(R.prop('y'))]);
-  const attackPatternsData = attackPatterns.length > 0 ? sortByY(attackPatternsFakeData) : [];
-
   // Exercises
   const [exercises, setExercises] = useState<EndpointStore[]>([]);
   const searchPaginationInput = {
```
Comment (guillaumejparis) [inline openbas-front/src/admin/components/Dashboard.tsx:74]: add DESC

**Comment by RomuDeuxfois** [comment]
Homogenize queries: base on 180 days for creation date and start date is not null.

**Pull request merged**